### DISCUSSION TITLE
Resetting form fields doesn't appear to be working

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#10354] [Help] [answered]
[Accotding to the docs](https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component#resetting-a-forms-data), we can reset a form with just `$this->form->fill()`. The component below is using this but the fields with the form are not being reset. Unsure if this is a bug/documentation issue/my code.

```php
class TestComponent extends Component implements HasForms
{
    use InteractsWithForms;

    public ?array $data;
    public string $message = '';

    public function mount()
    {
        $this->form->fill();
    }

    public function form(Form $form): Form
    {
        return $form
            ->statePath('data')
            ->schema([
                TextInput::make('testing')
                    ->live()
            ]);
    }

    public function display()
    {
        $this->message = $this->data['testing'];
        $this->form->fill();
    }

    public function render()
    {
        return view('my.test-component');
    }
}
```
```blade
<div>
    {{ $this->form }}
    <button type='button' wire:click='display'>Reset</button>
    {{ $this->message }}
</div>
```
There is a similar issue [here](https://github.com/filamentphp/filament/discussions/8979) but the author did not elaborate. Anyone else come across this? Any temp workarounds? Any help appreciated!

**Comment by ShaungBhone** [comment]
```php
public ?array $data = [];
```

**Reply by titantwentyone** [reply]
Have tried this and setting default data but same issue.

**Comment by titantwentyone** [comment, marked as answer]
Old project which was importing Alpine via the head. Removing this fixed the issue!
